crates/uv-python/src/installation.rs

```diff
@@ -298,28 +332,16 @@ impl FromStr for PythonInstallationKey {
             PythonInstallationKeyError::ParseError(key.to_string(), format!("invalid libc: {err}"))
         })?;
 
-        let [major, minor, patch] = version
-            .splitn(3, '.')
-            .map(str::parse::<u8>)
-            .collect::<Result<Vec<_>, _>>()
-            .map_err(|err| {
-                PythonInstallationKeyError::ParseError(
-                    key.to_string(),
-                    format!("invalid Python version: {err}"),
-                )
-            })?[..]
-        else {
-            return Err(PythonInstallationKeyError::ParseError(
+        let version = PythonVersion::from_str(version).map_err(|err| {
```

Comment:
Annoyed that we parse here then unparse and parse later but it seemed painful otherwise.

Reply:
The alternative is to hand-roll the parser like I do in the download metadata Python script.